.github/workflows/mac.yml

```diff
@@ -19,7 +19,7 @@ env:
   python_version: 3.11
 
 jobs:
-  MacOS_C_API_Packaging_CPU_x86_64:
+  x86_64-Xcode14:
```

Comment:
Why do you change this name?

Reply:
I just want to align with the other names in the job (eg. arm64-xcode16). is there a reason to keep them inconsistent?

Reply:
Nothing. This job was converted from an Azure DevOps pipeline, so the name kept unchanged.